### DISCUSSION TITLE
Questions about wiki.js 'theme' customization

### DISCUSSION THREAD

**Discussion by bkrepp-ctps** [requarks/wiki#6397] [Help / Questions]
Hi all,
I'm looking into what would be involved in doing some customization of the
appearance of our wiki.js sites, and have some questions.

I noiced that there is a 'Theme' section in the wiki.js 'Admin' interface.
This allows:

1. the override of CSS rules 'built in' to wiki.js
2. the injection of HTML in the <head> element of each page
3. the injection of HTML at the end of the <body> element of each page

I also noticed that wiki.js loads the jQuery library.

With all this in mind, I set out to try to make my first customization:
replace the string 'Wiki.js' that appears in the 'header bar' of our wikis.

Poking around inside the Google Developer Console, I noted that this element
can be selected by the jQuery rule:

	$('div.v-toolbar__title.mx-1 span.subheading').text('My wiki');
	
So, I thought that injecting the following snippet of HTML (containing
JS code) just before the close of the <head> tag should do the trick:

<script>
$(document).ready(function() {
	$('div.v-toolbar__title.mx-1 span.subheading').text('My wiki');
});
<script/>

I entered this change in the Admin interface's 'Theme' section, and 
applied it. I then loaded one of the pages of my wiki,... and saw
no change.

Did I go about doing this correctly, or not? If not, can someone provide
some guidance on how to proceed?

It occurred to me that there may be some caching going on which prevents
my 'seeing' the change. I flushed my browser's cache, and still saw no
change. Is there a server-side cache that also needs to be flushed?
If so, how does one go about doing this?

Thanks for any help you can offer with these questions.
-- Ben Krepp (bkrepp@ctps.org)

**Comment by NGPixel** [comment]
Not sure how you assume Wiki.js loads jQuery as that's definitely not the case.

You can change the title of your wiki under Admin > General.

**Comment by bkrepp-ctps** [comment]
Hi Nicolas, 

Thanks for your feedback. 

It certainly _appeared_ that the jQuery library was being loaded - in the Google Dev console I was able to select page elements using jQuery-style selectors, e.g., $('div.v-toolbar__title.mx-1 span.subheading'), so I made that assumption. My apologies for having jumped to an incorrect conclusion.

Given your reply, I conclude that wiki.js is _not_ loading the jQuery library, and consequently if I want/need to use it, I'll need to have the relevant <script> tag injected in the page <head>. Is this correct?

Regarding changing page styling, is there any documentation/guide on how to go about doing this?
In particular, we'd like to get rid of the "underlining" that's applied to most (all?) of the header elements, e.g., <h1>.

The relevant CSS rule for the <h1> element appears to be:

`
.v-main .contents h1:after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    background: linear-gradient(90deg,#1976d2,rgba(25,118,210,0));
    border-radius: 3px;
}
`
and we'd want to remove the "background".

Would all that's involved be inserting a CSS rule of our own for the '.v-main .contents h1:after' selector that doesn't include the line for the background property?

Thanks also for the tip on how to change the title of my Wiki. I also note that the Admin > General page allows one to change the 'logo URL' - which would have been my next question.

Thanks again and best regards,
Ben

**Reply by bkrepp-ctps** [reply]
Any suggestions on the best practice to use to remove the "underlining" that is displayed after most (all?) of the header elements, e,g., the <h1> elements?
Is it as simple as, for example, overriding the .v-main .contents h1:after rule with one of our own that simply omits the "background" part of the rule?
We're generally pretty happy with the default styling used by wiki.js, but have gotten quite a few complaints about the "underlining" after the header elements, and would like to be able to address these.
Thanks for any help you can  offer.
Best regards,
Ben
